jwthenticator/tokens.py

```diff
@@ -102,10 +104,10 @@ async def check_refresh_token_exists(self, refresh_token: str) -> bool:
         """
         Check if a refresh token exists in DB.
         """
-        async with self.session_factory() as session:
-            if await session.query(RefreshTokenInfo).filter_by(token=refresh_token).count() == 1:
-                return True
-        return False
+        async with self.async_session_factory() as session:
+            query = select(RefreshTokenInfo).where(RefreshTokenInfo.token == refresh_token)
```

Comment:
Consider query the count instead of query the whole row (and same for the rest of the places where you only checking the length of the results)